### PR TITLE
ci: add script to automate version release

### PR DESCRIPTION
**Pull request by tcrasset**
- ci: create script to bump version
- ci: create script to release a new version
- chore: run flake8 on code outside avatars package

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,6 +36,7 @@ lint-fix: ## Fix linting
 
 lint: ## Lint source files
 	poetry run bandit -r avatars -c bandit.yaml
+	poetry run flake8 .
 .PHONY: lint
```
Comment (tas17) [inline Makefile:42]: Add it in makefile ?

File: release.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        result = subprocess.call(command)
+        if result != 0:
+            raise typer.Exit(result)
+
+
+@app.command()
+def bump_version(
+    bump_type: BumpType = typer.Option(BumpType.PATCH, case_sensitive=False)
+) -> Any:
+    _bump_version(bump_type)
```
Comment (tas17) [inline release.py:144]: IMO not to be integrated

File: release.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+def commit_and_tag() -> None:
+    match = get_version_from_file(PYPROJECT_TOML)
+    if not match:
+        raise ValueError(f"Could not read version from file {PYPROJECT_TOML}.")
+
+    new_version = ".".join(match.groups())
+
+    commands = [
+        ("git", "add", str(PYPROJECT_TOML), str(INIT_PY), str(CHANGELOG)),
+        ("git", "commit", "-m", f"chore: release version {new_version}"),
```
Comment (tas17) [inline release.py:116]: If commit fail, unstage :)

File: release.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        ("git", "tag", "-s", f"{new_version}", "-m", f"Release version {new_version}"),
+    ]
+
+    for command in commands:
+        typer.echo(" ".join(command))
+        result = subprocess.call(command)
+        if result != 0:
+            raise typer.Exit(result)
```
Comment (tas17) [inline release.py:124]: Setting pipefail (`set -o pipefail` I think) would be the best practice in bash. Is it possible with typer ?

**Review by tas17 — COMMENTED** [review]
Nice first base! No major comment :)

File: release.py (hunk 4 above)
Comment (tcrasset) [reply release.py:124]: Hmm, I don't think it's a `typer` thing, we are not really doing any bash scripting here, we are just executing shell commands.

**Pull request merged**